node/service/src/service.rs

```diff
@@ -262,9 +266,39 @@ async fn start_node_impl(
 		})
 		.await?;
 
+	// Start off-chain workers if enabled
+	if parachain_config.offchain_worker.enabled {
+		log::info!("OFFCHAIN WORKER is Enabled!");
+		let offchain_workers =
+			sc_offchain::OffchainWorkers::new(sc_offchain::OffchainWorkerOptions {
+				runtime_api_provider: client.clone(),
+				is_validator: validator,
+				keystore: Some(params.keystore_container.keystore()),
+				offchain_db: backend.offchain_storage(),
+				transaction_pool: Some(OffchainTransactionPoolFactory::new(
+					transaction_pool.clone(),
+				)),
+				network_provider: network.clone(),
+				enable_http_requests: true,
+				custom_extensions: |_| vec![],
+			});
+
+		// Spawn a task to handle off-chain notifications.
+		// This task is responsible for processing off-chain events or data for the blockchain.
+		task_manager.spawn_handle().spawn(
+			"offchain-workers-runner",
+			"offchain-work",
+			offchain_workers.run(client.clone(), task_manager.spawn_handle()).boxed(),
```

Comment:
Consider cloning the client once and using it in both places.

Reply:
Not possible based on how Rust works!